include/triton/Dialect/TritonNvidiaGPU/Transforms/TMAUtilities.h

```diff
@@ -82,11 +82,15 @@ updateEncodingForShape(Operation *op, gpu::SharedEncodingTrait encoding,
 
     auto rank = tensorType.getRank();
     auto oldOrder = swizEnc.getOrder();
-    assert(oldOrder.size() <= rank);
     SmallVector<unsigned> order;
     for (int i = 0; i + oldOrder.size() < rank; ++i)
       order.push_back(rank - i - 1);
-    order.append(oldOrder.begin(), oldOrder.end());
+    for (int i = 0; i < oldOrder.size(); ++i) {
+      // If it is a rank-reducing load, we need to drop the last dimensions.
+      if (oldOrder[i] >= rank)
+        continue;
+      order.push_back(oldOrder[i]);
+    }
```

Comment:
NIT: could be simplified with `ArrayRef oldOrder`.
```suggestion
    if (oldOrder.size() > rank)
        oldOrder = oldOrder.drop_front(rank - oldOrder.size());
    order.append(oldOrder.begin(), oldOrder.end());
```

Reply:
is that equivalent? This assumes that the higher dim are first?

Reply:
Hrm I think we should always have order as c-contiguous otherwise the TMA lowering would be incorrect.

Reply:
well technically since the higher dim is 1 I don't think it matters?

Reply:
but you are right, in general it would be in order, do you prefer it that way?